#!/usr/bin/env python
PACKAGE = "autosail"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()
gen.add("sail_limit", int_t, 0, "Rudder limit in degrees where 0 is straight", 80, 0, 90)
pid_group = gen.add_group("pid_coefficients")
pid_group.add("kp", double_t, 0, "Proportional gain", 1, 0, 100)
pid_group.add("ki", double_t, 0, "Integral gain (reset)", 0, 0, 1000)
pid_group.add("kd", double_t, 0, "Derivative gain (preact)", 0, 0, 1000)
gen.add("enable_auto_tilt", bool_t, 0, "Enable/Disable Automatic Tilt Based on IMU", True)
exit(gen.generate(PACKAGE, "sail_controller", "SailController"))
